src/pd/async/util.rs

```diff
@@ -74,34 +82,52 @@ impl LeaderClient {
     }
 
     // Re-establish connection with PD leader in synchronized fashion.
-    pub fn reconnect(&self) {
-        let start = Instant::now();
-        let ret = try_connect_leader(&self.inner.rl().members);
+    pub fn reconnect(&self) -> PdFuture<()> {
+        let start;
+        let sleep;
+        let ret = {
+            let inner = self.inner.rl();
+            sleep = inner.timer
```

Comment:
Actually, its not. It only return a `Sleep` future, caller decides whether `wait` it or not, the first connection happens in tikv-server.rs which does not preform `wait`.